### PR TITLE
Choose the familiar + equip combo that results in the heaviest weight

### PR DESCRIPTION
**Pull request by chaucerbao**
The day after you merged the piccolo check, my garden popped out an amulet coin...

I couldn't find a way to dynamically detect +familiar weight equipment yet, so I just added another item to the list.  Lemme know if there's a better way.

### CODE REVIEW

**Comment by Pantocyclus** [comment]
Consider implementing a function in `lib.ts` like
```
export function bestFamiliarEquip(checkedFamiliar: Familiar): Item {
    const currentFamiliar = myFamiliar();
    useFamiliar(checkedFamiliar); // canEquip appears to check if the item can be equipped on the current familiar, but needs testing
    const validEquips = $items``.filter((it) => have(it) && numericModifier(it, "Familiar Weight") > 0 && canEquip(it));
    useFamiliar(currentFamiliar);
    if (validEquips.length === 0)
        return $item.none;
    const bestEquip = validEquips.reduce((a, b) => numericModifier(a, "Familiar Weight") > numericModifier(b, "Familiar Weight") ? a : b);
    return bestEquip;
}
```
(Which may be better than just returning the maximum value outright, since we will need to know what to equip if we determine that doing so is the best course of action)

There is a caveat here that we are now choosing the heaviest familiar prior to considering what equipments it could use (there could potentially be a familiar with a lower unmodified weight, but having access to an equipment with a large modifier, but `chooseHeaviestFamiliar()` only returns the familiar with the highest unmodified weight, leading to some suboptimal results). Perhaps this edge case can be solved in a future update.

**Comment by chaucerbao** [comment]
I'll try this after RO

File: src/lib.ts [hunk 1]
```diff
@@ -494,6 +498,36 @@ export function burnLibram(saveMp: number): void {
...
+
+      return {
+        familiar,
+        equip: bestEquip,
+        expectedWeight: familiarWeight(familiar) + numericModifier(bestEquip, "Familiar Weight"),
+      };
+    })
+    .reduce((a, b) => (a.expectedWeight > b.expectedWeight ? a : b));
+}
+
```
Comment (Pantocyclus) [inline src/lib.ts:530]: This is nice. We can deprecate `chooseHeaviestFamiliar()` in `outfit.ts`.
Also, and I just noticed that it wasn't in `chooseHeaviestFamiliar()` either, we should add `!excludedFamiliars.includes(familiar.id)` to the filter if the user so wishes to not use said familiars

Comment (chaucerbao) [reply src/lib.ts]: Done

**Comment by chaucerbao** [comment]
> I'll try this after RO

Seems to work.

**Pull request merged**